### ISSUE TITLE
Buffer for a file in a project hidden from helm-mini by perspective if helm-persp-switch-project is used while buffer active

### ISSUE THREAD

**Issue by jesuspc** [syl20bnr/spacemacs#8073]
#### Description :octocat:
Buffer not shown in helm-mini anymore when using switch perspective under
special conditions.

#### Reproduction guide :beetle:
- Start Emacs
- Navigate to a file inside a project without using perspective, for instance
  using helm-mini Recentf and open a buffer for it. I've reproduced the issue with Ruby, JS and Haskell files
- Use helm-persp-switch-project to switch to the project to which the file
  belongs to
- Open helm-mini. The buffer for that file will not appear (not expected), even
  after navigating to the file manually. Other buffer-related actions involving
  that buffer do not work neither. In particular SPC-TAB (last buffer). The
  buffer does appear in iBuffer.

*Observed behaviour:* :eyes: :broken_heart:
- The buffer is not shown in helm-mini
- Last buffer never goes back to the buffer containing the file

*Expected behaviour:* :heart: :smile:
- The buffer should be shown in helm-mini
- Last buffer should work normally and be able to show the buffer containing the
  file if needed

#### System Info :computer:
- OS: darwin
- Emacs: 25.1.1
- Spacemacs: 0.200.5
- Spacemacs branch: master (rev. 664ba6a)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(octave slack yaml restclient auto-completion spotify better-defaults emacs-lisp osx fasd docker slack private-layer git github markdown pandoc shell search-engine org
        (shell :variables shell-default-height 30 shell-default-position 'bottom shell-default-shell 'eshell)
        syntax-checking version-control
        (haskell :variables haskell-enable-ghc-mod-support nil haskell-process-type 'stack-ghci haskell-completion-backend 'intero haskell-process-args-stack-ghci
                 '("--ghc-options=-ferror-spans" "--with-ghc=intero")
                 hindent-style "johan-tibell" haskell-stylish-on-save t)
        ruby ruby-on-rails elixir erlang html javascript nlinum
        (osx :variables osx-use-option-as-meta nil)
        ranger ibuffer)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by gone** [comment]
I can replicate this - emacs version GNU Emacs 26.2 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2019-04-12
Spacemacs version 0.200.13

**Comment by duianto** [comment]
A recent update on the `develop` branch made changes to how buffers are added to perspectives:
spacemacs-layouts: Fix adding buffers to new persp #12287
